NK_Diplom_1

natyko opened the pull request:
Update project structure 
Add test cases
![Uploading Screenshot 2025-04-06 at 8.01.50 PM.png…]()

Commit:
add test cases
update project structure

irisqul commented on tests/burger_test.py:
```diff
@@ -0,0 +1,141 @@
+class TestBurger:
+    """Tests for the Burger class."""
+
+    @pytest.fixture
```
Нужно исправить: перенести фикстуры в conftest

natyko replied:
done

irisqul commented on tests/burger_test.py:
```diff
@@ -0,0 +1,141 @@
+    def test_burger_creation(self, empty_burger):
+        """Test that a burger can be created."""
+        # Assert
+        assert empty_burger.bun is None
```
Нужно исправить: мы тестируем не атрибуты экземпляров класса или наполнение базы данных, а методы приложения. Необходимо добавить шаги в которых будут использованы методы и проверить результат.

natyko replied:
done

irisqul commented on tests/database_test.py:
```diff
@@ -0,0 +1,75 @@
+    def test_database_creation(self, database):
+        """Test that a database is initialized with the correct data."""
+        # Assert
+        assert len(database.buns) == 3
```
Нужно исправить: мы тестируем не атрибуты экземпляров класса или наполнение базы данных, а методы приложения. Необходимо добавить шаги в которых будут использованы методы и проверить результат.
 
 В тестах в этом файле нет шагов и тестовых сценариев, они не нужны

natyko replied:
done

irisqul commented on tests/praktikum_test.py:
```diff
@@ -0,0 +1,41 @@
+        mock_database.available_ingredients.assert_called_once()
+        mock_burger.set_buns.assert_called_once_with(mock_buns[0])
+        assert mock_burger.add_ingredient.call_count == 4
+        mock_burger.move_ingredient.assert_called_once_with(2, 1)
```
нужно исправить: тест неатомарный

natyko replied:
done

irisqul reviewed (commented):
✅  Сильные стороны:

* Код запускается без ошибок
* Тестовая архитектура соответствует изученным практикам написания автотестов

Нужно исправить: пожалуйста добавь в пуллреквест отчет о проценте покрытия кода тестами

Commit:
update test cases, fixed comments

natyko commented:
<img width="606" height="462" alt="Screenshot 2025-08-26 at 10 12 40 PM" src="https://github.com/user-attachments/assets/9d0f25e3-4b31-4825-ab9c-d13a862b9207" />
 Updated files, addressed comments

Commit:
fixed comments

Commit:
cut unused imports

Commit:
add conftest.py
